### PR TITLE
DS-42 restore schema validation twig dump

### PR DESCRIPTION
**Pull request by remydenton**
## Jira

https://pegadigitalit.atlassian.net/browse/DS-42

## Summary

Restores both schema validation and twig dump function in Pattern Lab.  Also fixes numerous schema violations.

## Details

#1802 disabled twig debugging in the name of performance.  That broke several useful things, including schema validation and the ability to use the twig [dump](https://twig.symfony.com/doc/2.x/functions/dump.html) function.

This PR does the following:
1. Updates the pattern lab patches to allow twig debugging to be enabled
2. Enables twig debugging by default in docs-site/config/config.yml
3. Fixes all the schema errors that have been added in while schema validation was broken

The only downside is that Item 2 decreases performance somewhat.  I found my re-compile times to be about 10 seconds before these changes and about 15 seconds after them.  I think that's worth it to be able to catch your bugs immediately and use the twig debug function, but we could potentially turn it off conditionally (e.g. maybe only enable it on Travis?).  As it stands, developers could also manually update that line locally to disable twig debugging while working locally.  I'm open to suggestions on this.

#### Confirm that schema validation and twig dump are working
- Checkout this branch locally and run `yarn setup`
- Add an intentional schema violation in pattern lab, such as:
```
{% include "@bolt-components-button/button.twig" with {
  text: 'foo',    
  size: 'not a valid size',
} only %}
```
- Run `yarn start`.  Building should fail and you'll get an error message about your schema validation in the cli.
- Remove the schema validation and run `yarn start` again and wait for PL to load.
- Add the schema validation back again after PL has loaded.  Confirm that there is again a schema validation error in the cli.  - Confirm that removing the schema validation results in a successful build without the need to restart
- Add the snippet `{{ dump('hello world') }}` to any twig file in pattern lab (e.g. 05-button.twig).  Confirm that it outputs `string(2) "hi"` to the page where this twig file is rendered.  You can also test dumping objects, arrays, etc.

#### Review all code changes for sanity/logic
Most of the commits in this PR fix problems in the schema.  Each one is small, so the ideal thing to do is look at each one and confirm that the fix makes sense.  You can also check the corresponding page in pattern lab to make sure things look as expected.

### CODE REVIEW

**Review by colbytcook — APPROVED** [review]

**Review by danielamorse — APPROVED** [review]
Verified all of the above. Note: I had to run `yarn clean` then `yarn start` before the validation kicked in.

Glad to have validation working again, especially seeing how many errors piled up in the short time we had it turned off.

I left a couple inline comments just checking my understanding of the code changes.

File: packages/components/bolt-card-replacement/src/card-replacement-body/_card-replacement-body.twig [hunk 1]
```diff
@@ -1,9 +1,5 @@
 {% set schema = bolt.data.components["@bolt-components-card-replacement"].schema %}
 
-{% if enable_json_schema_validation %}
```
[the remark this answers is not in the document]
Comment (remydenton) [reply packages/components/bolt-card-replacement/src/card-replacement-body/_card-replacement-body.twig:3]: That was my justification @danielamorse-- seems like validation doesn't make that much sense for internal templates.  And if it does, should it really be the _same_ validation as we have for the parent component (which includes lots of props that wouldn't be valid here)?

In actuality, this my shortcut to having to fix the logic for [the theme prop in card-replacement.twig](https://github.com/boltdesignsystem/bolt/blob/d8e2b0672604409bc570d54d7515c2d67d7a0c47/packages/components/bolt-card-replacement/src/card-replacement/card-replacement.twig#L64).  That transforms theme into a boolean, which then causes validation in `_card-replacement-body.twig` to fail for that prop, even though the `theme` prop isn't even used in that file.

Ultimately, I'm torn on this.  I could fix the logic for `theme` if you think that's a better approach.

File: packages/components/bolt-share/src/share.twig [hunk 2]
```diff
@@ -79,7 +79,7 @@
     {% set icon_before %}
       {% include '@bolt-components-icon/icon.twig' with {
         name: source_icon,
-        size: size == 'medium' ? 'small',
```
[the remark this answers is not in the document]
Comment (remydenton) [reply packages/components/bolt-share/src/share.twig:82]: Great question.  Is the default icon size `xsmall`?  That wasn't even [an option in the schema previously](https://github.com/boltdesignsystem/bolt/pull/2032/files#diff-cc2fc812532d0d23913448c0f13591bd8a6cd9850aa94117ae2970effa60a5c9R178).   In any case, we should add a default value for icon size.

File: packages/components/bolt-card-replacement/src/card-replacement-body/_card-replacement-body.twig (hunk 1 above)
[the remark this answers is not in the document]
Comment (danielamorse) [reply packages/components/bolt-card-replacement/src/card-replacement-body/_card-replacement-body.twig:3]: Right, if we're validating a child template we should at least be validating _that_ template's props not the parent's props. I'm find with this approach for now 👌 

I'd wait to fix the `theme` logic when/if we do a bigger component overhaul, or perhaps when @cjwhitedev is upgrading the JS/Twig conventions.

File: packages/components/bolt-share/src/share.twig (hunk 2 above)
[the remark this answers is not in the document]
Comment (remydenton) [reply packages/components/bolt-share/src/share.twig:82]: Icon size looks the same to me between this branch and master.  Are you seeing a difference somewhere besides the following two links @danielamorse?

https://boltdesignsystem.com/pattern-lab/?p=components-share-size-variations
https://feature-ds-42-restore-schema-validation-t.boltdesignsystem.com/pattern-lab/?p=components-share-size-variations

[the remark this answers is not in the document]
Comment (mikemai2awesome) [reply packages/components/bolt-share/src/share.twig:82]: Yup, all good. This is for Menu display, we can always set it to small.

**Review by mikemai2awesome — APPROVED** [review]

File: packages/components/bolt-share/src/share.twig (hunk 2 above)
[the remark this answers is not in the document]
Comment (danielamorse) [reply packages/components/bolt-share/src/share.twig:82]: @remydenton, I wasn't seeing a difference. I was just following the logic and assuming that previously icon would sometimes get set to `size: null`. If @mikemai2awesome is good with this, I'm good too 👌

**Pull request merged**